design-proposals/export-import/volume-export.md

```diff
@@ -0,0 +1,161 @@
+# Overview
+Proposal for exporting KubeVirt Virtual Machine volumes from a Kubernetes cluster
+
+## Motivation
+There are currently multiple ways to import/initialize Virtual Machine volumes.  But there is no easy and efficient method to get volume data out of the cluster.
+
+## Goals
+* Define VirtualMachineExport API
+* Introduce virtctl export command
+
+## Non Goals
+* Implementation details of internal components (servers/proxies/controllers/etc)
+
+## User Stories
+* As a KubeVirt user, I would like to export a Virtual Machine volume for local debugging/analysis.
+* As a KubeVirt user, I would like to export a Virtual Machine volume as part of migrating a Virtual Machine to a different cluster.
+* As a KubeVirt user, I would like to export a Virtual Machine volume to serve as an offsite backup.
+
+## Repos
+* [KubeVirt](https://github.com/kubevirt/kubevirt)
+
+# Design
+
+## VirtualMachineExport API
+
+The VirtualMachineExport API allows users to download a Virtual Machine volumes from URLs.
+
+### User Flow
+
+1.  Create a VirtualMachineExport resource
+2.  Wait for the `Ready` condition of the VirtualMachineExport to be `True`
+3.  Download a Virtual Machine volume in the desired format listed under `status.links.volumes`
+
+### Export Formats
+
+| Volume Content Type | Export Format |
+| ------------------- | ------------- |
+| kubevirt            | raw           |
+| kubevirt            | tar.gz        |
+| archive             | dir           |
+| archive             | tar.gz        |
+
+#### raw
+
+The http endpoint supports random access to a raw disk image.
+
+#### tar.gz
+
+The http endpoint is a link to a `tar.gz` streaming download.
+
+#### dir
+
+The http endpoint is a golang `http.FileServer` serving up the root of the filesystem.
+
+### Export a VirtualMachine
+
+``` yaml
+apiVersion: export.kubevirt.io/v1alpha1
+kind: VirtualMachineExport
+metadata:
+    name: export-vm1
+spec:
+    source:
+        apiGroup: kubevirt.io
+        kind: VirtualMachine
+        name: vm1
+```
```

Comment:
Idea (which may be postponed for phase 3 of the implementation): add an optional `target` section, which the vm owner can use to specify an S3 bucket and a secret to access it. This would allow me to export (an archive? of) the VM to the cloud, without having to go through my laptop.